leapsectz/leapsectz_test.go

```diff
@@ -58,3 +58,118 @@ func TestParse(t *testing.T) {
 		t.Errorf("wrong leap second count")
 	}
 }
+
+func TestParseV2(t *testing.T) {
+	byteData := []byte{
+		'T', 'Z', 'i', 'f', // magic
+		'2', 0x00, 0x00, 0x00, // version
+		0x00, 0x00, 0x00, 0x00, // pad
+		0x00, 0x00, 0x00, 0x00, // pad
+		0x00, 0x00, 0x00, 0x00, // pad
+		0x00, 0x00, 0x00, 0x00, // UTC/local
+		0x00, 0x00, 0x00, 0x00, // standard/wall
+		0x00, 0x00, 0x00, 0x02, // leap
+		0x00, 0x00, 0x00, 0x00, // transition
+		0x00, 0x00, 0x00, 0x00, // local tz
+		0x00, 0x00, 0x00, 0x00, // characters
+		0x04, 0xb2, 0x58, 0x00, // leap time
+		0x00, 0x00, 0x00, 0x01, // leap count
+		0x05, 0xa4, 0xec, 0x01, // leap time
+		0x00, 0x00, 0x00, 0x02, // leap count
+		'T', 'Z', 'i', 'f', // magic
+		'2', 0x00, 0x00, 0x00, // version
+		0x00, 0x00, 0x00, 0x00, // pad
+		0x00, 0x00, 0x00, 0x00, // pad
+		0x00, 0x00, 0x00, 0x00, // pad
+		0x00, 0x00, 0x00, 0x00, // UTC/local
+		0x00, 0x00, 0x00, 0x00, // standard/wall
+		0x00, 0x00, 0x00, 0x02, // leap
+		0x00, 0x00, 0x00, 0x00, // transition
+		0x00, 0x00, 0x00, 0x00, // local tz
+		0x00, 0x00, 0x00, 0x00, // characters
+		0x00, 0x00, 0x00, 0x00, // leap time (first 32 bits)
+		0x04, 0xb2, 0x58, 0x00, // leap time (last 32 bits)
+		0x00, 0x00, 0x00, 0x01, // leap count
+		0x00, 0x00, 0x00, 0x00, // leap time (first 32 bits)
+		0x05, 0xa4, 0xec, 0x01, // leap time (last 32 bits)
+		0x00, 0x00, 0x00, 0x02, // leap count
+	}
+
+	r := bytes.NewReader(byteData)
+
+	ls, e := parseVx(r)
```

Comment:
Testing "positive" is good. Could you maybe pass a corrupted file to make sure "errors" are triggered?
2 scenarios:
`string(magic) != "TZif"`
`if n, _ := r.Read(p); n != 16 {`
Both must return `errBadData`

This will give you an extra coverage.

Reply:
added test with bad input data